consensus/src/state_computer.rs

```diff
@@ -327,6 +327,22 @@ impl StateComputer for ExecutionProxy {
             LogicalTime::new(target.ledger_info().epoch(), target.ledger_info().round());
         let block_timestamp = target.commit_info().timestamp_usecs();
 
+        // The `target` might be the end of the current epoch, in that case if we already
+        // (pre-)committed that version, state sync won't be able to sync to it, because
+        // it always requests `pre_committed_version + 1` AND WITHIN THE CURRENT EPOCH.
+        // To get around that, we try to identify the situation and avoid switching to state-sync
+        if self.executor.committed_block_id() == target.commit_info().id() {
+            info!("Epoch ending already committed, skip state sync.");
```

Comment:
this is not only called when sync_to epoch boundary, also isn't the statement at L351 covers this case maybe just move it up?

Reply:
updated